### DISCUSSION TITLE
Vertex AI - Error: APIs that require a project ID cannot use an API key

### DISCUSSION THREAD

**Discussion by Niqnil** [danny-avila/LibreChat#2570] [Troubleshooting] [answered]
I am trying to use gemini through Vertex AI. I followed instructions at https://docs.librechat.ai/install/configuration/ai_setup.html#2-create-a-service-account-with-vertex-ai-role and have auth.json.

If I add auth.json to /LibreChat/api/data/auth.json and leave `GOOGLE_KEY=` empty in .env, Google does not appear in the librechat provider dropdown list.

Here's my current set up and the error log:

I left the Google API key empty when adding at frontend. Should I fill it with API key from Google AI studio?
![screenshot](https://i.ibb.co/ZT7nb4N/image.png "screenshot")

.env
```
GOOGLE_KEY=user_provided
GOOGLE_MODELS=gemini-1.5-pro-preview-0409,gemini-1.0-pro-vision-001,gemini-pro
```

error log
```
{"level":"error","message":"[handleAbortError] AI response error; aborting request: APIs that require a project ID cannot use an API key","stack":"Error: APIs that require a project ID cannot use an API key\n    at ApiKeyGoogleAuth.getProjectId (/app/node_modules/@langchain/google-common/dist/auth.cjs:56:15)\n    at ChatConnection.buildUrlVertex (/app/node_modules/@langchain/google-common/dist/connection.cjs:171:45)\n    at ChatConnection.buildUrl (/app/node_modules/@langchain/google-common/dist/connection.cjs:181:29)\n    at ChatConnection._request (/app/node_modules/@langchain/google-common/dist/connection.cjs:51:32)\n    at ChatConnection.request (/app/node_modules/@langchain/google-common/dist/connection.cjs:186:37)\n    at ChatVertexAI._streamResponseChunks (/app/node_modules/@langchain/google-common/dist/chat_models.cjs:173:56)\n    at _streamResponseChunks.next (<anonymous>)\n    at ChatVertexAI._streamIterator (/app/node_modules/@langchain/core/dist/language_models/chat_models.cjs:82:34)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)"}
```

Version: v0.7.1
I am running with docker compose.

**Comment by danny-avila** [comment, marked as answer]
Are you using docker? If so, you need to also mount the volume in docker-compose.override.yml

```yaml
version: '3.4'

services:
  api:
    volumes:
    - type: bind
      source: ./api/data/auth.json
      target: /app/api/data/auth.json
```

Basically, if you at least have the `auth.json` file, you don't need to define `GOOGLE_KEY` and it will show

![image](https://github.com/danny-avila/LibreChat/assets/110412045/05f5d71a-9b3f-4c8e-918e-1217dbfd43f3)

**Reply by Niqnil** [reply]
Yes I am using docker.
After mounting the volume in docker-compose.override.yml, I don't have to set the key or import auth.json in the frontend. It doesn't seem to matter if I set `# GOOGLE_KEY=` or `GOOGLE_KEY=`(empty) in .env. 
The models appear. But I still get the error.

![screenshot](https://i.ibb.co/XttF2RZ/image.png "screenshot")

```
{"level":"error","message":"[handleAbortError] AI response error; aborting request: APIs that require a project ID cannot use an API key","stack":"Error: APIs that require a project ID cannot use an API key\n    at ApiKeyGoogleAuth.getProjectId (/app/node_modules/@langchain/google-common/dist/auth.cjs:56:15)\n    at ChatConnection.buildUrlVertex (/app/node_modules/@langchain/google-common/dist/connection.cjs:171:45)\n    at ChatConnection.buildUrl (/app/node_modules/@langchain/google-common/dist/connection.cjs:181:29)\n    at ChatConnection._request (/app/node_modules/@langchain/google-common/dist/connection.cjs:51:32)\n    at ChatConnection.request (/app/node_modules/@langchain/google-common/dist/connection.cjs:186:37)\n    at ChatVertexAI._streamResponseChunks (/app/node_modules/@langchain/google-common/dist/chat_models.cjs:173:56)\n    at _streamResponseChunks.next (<anonymous>)\n    at ChatVertexAI._streamIterator (/app/node_modules/@langchain/core/dist/language_models/chat_models.cjs:82:34)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)"}
```

**Reply by Niqnil** [reply]
While reading the debug log to try to fix the error, I noticed the warning and decided to fix it. After updating .env, I tried Vertex Gemini again and it works now. Seems the warning was related and might have caused the error.

Thanks for the info on docker volume mount. I will mark that as answer.

```
warn: The `GOOGLE_API_KEY` environment variable is deprecated.
Please use the `GOOGLE_SEARCH_API_KEY` environment variable instead.
```

Relevant .env settings, in case it helps someone
```
# GOOGLE_KEY=
GOOGLE_MODELS=gemini-1.5-pro-preview-0409,gemini-1.0-pro-vision-001,gemini-pro
# GOOGLE_REVERSE_PROXY=

GOOGLE_SEARCH_API_KEY=redacted
GOOGLE_CSE_ID=redacted
```

**Reply by danny-avila** [reply]
awesome, thank you for sharing, sorry I wasn't much help here
